File: sorts/insertion/insertion.py
def insertion_sort(array):
    i, j = 0, 0
    isSmallest = True
    for i in range(1, len(array)):
        temp = array[i]
        isSmallest = True
        for j in range(i - 1, -1, -1):
            if array[j] > temp:
                array[j + 1] = array[j]
            else:
                array[j+1] = temp
                isSmallest = False
                break
        if isSmallest:
            array[j] = temp


array1 = [8, 3, 5, 9, 1, 2, 7]
insertion_sort(array1)
print(array1)
array2 = [9, 8, 7, 6, 5, 4, 3, 2 ,1]
insertion_sort(array2)
print(array2)
array3 = [9, 9, 9, 7, 5, 7, 3, 7, 3, 5, 3]
insertion_sort(array3)
print(array3)

File: sorts/insertion/insertion.js
function insertionSort(array) {
    var temp, i, j;
    for (i =  1; i < array.length; i++) {
        temp = array[i];
        for (j = i - 1; j >= 0; j--) {
            if (array[j] > temp) {
                array[j+1] = array[j];
            } else {
                break;
            }
            array[j] = temp;            
        }
    }
    prettyPrint = "";
    for (var i = 0; i < array.length; i++) {
        prettyPrint += array[i] + " "
    }
    console.log(prettyPrint);
}

myArray = [9, 8, 7, 6, 5, 4, 3, 2, 1];
insertionSort(myArray);
myArray = [8, 4, 9, 1, 3, 7, 6];
insertionSort(myArray);
myArray = [9, 9, 9, 7, 7, 5, 3, 5, 7, 5, 3, 3];
insertionSort(myArray);
myArray = [8, 3, 5, 9, 1, 2, 7];
insertionSort(myArray);
